src/WebJobs.Script/ScriptHostBuilderExtensions.cs

```diff
@@ -408,10 +415,30 @@ internal static void ConfigureApplicationInsights(HostBuilderContext context, IL
             // Initializing AppInsights services during placeholder mode as well to avoid the cost of JITting these objects during specialization
             if (!string.IsNullOrEmpty(appInsightsInstrumentationKey) || !string.IsNullOrEmpty(appInsightsConnectionString) || SystemEnvironment.Instance.IsPlaceholderModeEnabled())
             {
+                // Initialize ScriptTelemetryInitializer before any other telemetry initializers.
+                // This will allow HostInstanceId to be removed as part of MetricsCustomDimensionOptimization.
+                builder.Services.AddSingleton<ITelemetryInitializer, ScriptTelemetryInitializer>();
                 builder.AddApplicationInsightsWebJobs(o =>
                 {
                     o.InstrumentationKey = appInsightsInstrumentationKey;
                     o.ConnectionString = appInsightsConnectionString;
+
+                    if (!string.IsNullOrEmpty(context.Configuration[EnvironmentSettingNames.AppInsightsEventListenerLogLevel]))
+                    {
+                        if (Enum.TryParse(context.Configuration[EnvironmentSettingNames.AppInsightsEventListenerLogLevel], ignoreCase: true, out EventLevel level))
+                        {
+                            o.DiagnosticsEventListenerLogLevel = level;
+                        }
+                        else
+                        {
+                            throw new InvalidEnumArgumentException($"Invalid `{EnvironmentSettingNames.AppInsightsEventListenerLogLevel}`.");
+                        }
+                    }
+
+                    if (!string.IsNullOrEmpty(context.Configuration[EnvironmentSettingNames.AppInsightsAuthenticationString]))
+                    {
+                        o.TokenCredentialOptions = TokenCredentialOptions.ParseAuthenticationString(context.Configuration[EnvironmentSettingNames.AppInsightsAuthenticationString]);
```

Comment:
2 questions here:
- What are the requirements to use this? An Instrumentation key *or* a Conn string are required? 
- I see this parsing is in our own code in WebJobs -- are there docs on what this auth string is meant to look like? is it readily copy-able from somewhere in the portal?

Reply:
Yes, need either an IKey or connection string. 

[APPLICATIONINSIGHTS_AUTHENTICATION_STRING](https://msazure.visualstudio.com/One/_git/Mgmt-AppInsights-StatusMonitor?path=%2FAgents%2FCommon%2FApplicationInsights.Identity%2Freadme.md&_a=contents&version=GBmaster#) is owned by the ApplicationInsights team. I have used some of their logic/code for consistency (like parsing, validation and error scenarios). 

I will update our documentation on how to enable AAD auth, including details about the auth string.